tests/test_dataset.py

```diff
@@ -424,3 +434,89 @@ def test_yaak(dataset: Dataset) -> None:
             logger.error(msg := "invalid batch structure", batch=batch)
 
             raise AssertionError(msg)
+
+
+@pytest.mark.parametrize("dataset", [lf("yaak_hydra"), lf("yaak_pydantic")])
+def test_waypoints_yaak(dataset: Dataset) -> None:
```

Comment:
since this essentially validates some subset of batch values, should we just put it under the first `case` in `test_yaak`?